### ISSUE TITLE
Support pip additional file types

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#2214]
`setup.cfg` - https://setuptools.readthedocs.io/en/latest/setuptools.html#configuring-setup-using-setup-cfg-files

`tox.ini` - https://tox.readthedocs.io/en/latest/

**Comment by jmahlik** [comment]
Is it possible that the `setup.cfg` manager's code could be used to implement this? The `tox.ini` file is the same `ini` style format.  The deps are just in a separate section.

So basically, it would just be looking for `[testenv*]` sections and extracting the `deps =`.  So, just need to implement the a custom extract that looks for these sections?

https://tox.wiki/en/latest/config.html

There are some oddities like [dependency matrix](https://tox.wiki/en/latest/example/basic.html#depending-on-requirements-txt-or-defining-constraints) but it seems like the regex for `setup.cfg` could handle this.

It would not work for requirements.txt in the `tox.ini` deps, but the requirements manager already would handle those. 

https://tox.wiki/en/latest/example/basic.html#depending-on-requirements-txt-or-defining-constraints

**Comment by HonkingGoose** [comment]
Copy/pasting a relevant comment from the old PR: [^quote]

> Thanks for the ping. I've been busy as of late. I plan to move the idea forward. But given the PR adding tox.ini support would look quite different after a refactor of the setup.cfg manager, we could close this one and handle it in a separate PR.

@jmahlik Do you want to be assigned to this issue, so you can work on it?

[^quote]: https://github.com/renovatebot/renovate/pull/16824#issuecomment-1283982842

**Comment by jmahlik** [comment]
> Copy/pasting a relevant comment from the old PR: [1](#user-content-fn-quote-e66db3c55f29fc2702f1204b08e54f22)
> 
> > Thanks for the ping. I've been busy as of late. I plan to move the idea forward. But given the PR adding tox.ini support would look quite different after a refactor of the setup.cfg manager, we could close this one and handle it in a separate PR.
> 
> @jmahlik Do you want to be assigned to this issue, so you can work on it?
> 
> ## Footnotes
> 1. https://github.com/renovatebot/renovate/pull/16824#issuecomment-1283982842 [↩](#user-content-fnref-quote-e66db3c55f29fc2702f1204b08e54f22)

Yes, that works for me. Just took a look at porting over the ini parser from python to get more robust ini parsing that can handle a `tox.ini`.

**Comment by edgarrmondragon** [comment]
WIP:

- https://github.com/renovatebot/renovate/pull/43975
